libs/base/Control/WellFounded.idr

```diff
@@ -58,7 +59,7 @@ sizeAccessible x = Access (acc $ size x)
   where
     acc : (sizeX : Nat) -> (y : a) -> (size y `LT` sizeX) -> SizeAccessible y
     acc (S x') y (LTESucc yLEx')
-        = Access (\z, zLTy => acc x' z (lteTransitive zLTy yLEx'))
+        = Access (\z, zLTy => acc x' z (transitive {x = (S (size z))} {y = (size y)} {z = x'} zLTy yLEx'))
```

Comment:
Does specifying `transitive {rel = LT}` work here?
That'd be more economical.

If so we may want to redefine `LT` to be a newtype wrapper.
With `Cast (LT x y) (LTE (S x) y)` and `Cast (LTE (S x) y) (LT x y)`
it may be nice enough to use if we get improved inference from it.

Reply:
Simplified to use `{rel = LT}`